### ISSUE TITLE
Error in log when load balancer (F5) check /system/lbstatus

### ISSUE THREAD

**Issue by ismaelpuerto** [Graylog2/graylog2-server#1658]
Hi

We have F5 for groylogs inputs. The F5 checks string "200 OK" in /system/lbstatus. When F5 match this string, it send FIN packet after it receives the entire requested object. With this request, graylog write:

```
2015-12-28T12:49:47.787Z ERROR [NettyContainer] Uncaught exception during jersey resource handling
java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:64)
	at org.jboss.netty.channel.socket.nio.AbstractNioWorker.process(AbstractNioWorker.java:108)
	at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:337)
	at org.jboss.netty.channel.socket.nio.AbstractNioWorker.run(AbstractNioWorker.java:89)
	at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:178)
	at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108)
	at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42)
	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:176)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2015-12-28T12:49:47.789Z WARN  [DefaultChannelPipeline] An exception was thrown by a user handler while handling an exception event ([id: 0xfeea709c, /10.68.30.3:18919 :> /10.68.30.43:12900] EXCEPTION: java.io.IOException: Connection reset by peer)
java.lang.NullPointerException
	at org.graylog2.jersey.container.netty.NettyContainer.exceptionCaught(NettyContainer.java:412)
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:112)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
	at org.jboss.netty.handler.execution.ChannelUpstreamEventRunnable.doRun(ChannelUpstreamEventRunnable.java:43)
	at org.jboss.netty.handler.execution.ChannelEventRunnable.run(ChannelEventRunnable.java:67)
	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:176)
	at org.jboss.netty.handler.execution.MemoryAwareThreadPoolExecutor$MemoryAwareRunnable.run(MemoryAwareThreadPoolExecutor.java:606)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)'
```

More info: [https://support.f5.com/kb/en-us/solutions/public/11000/300/sol11359](https://support.f5.com/kb/en-us/solutions/public/11000/300/sol11359)

**Comment by mikkolehtisalo** [comment]
I managed to reproduce this on 1.3 with the following rough C program. 2.0-alpha5 seems to handle similar situation silently.

```
#include <stdio.h>
#include <string.h>
#include <sys/socket.h>
#include <arpa/inet.h>
#include <unistd.h>

int main(int argc , char *argv[])
{
	int socket_desc;
	struct sockaddr_in server;
	char *message , server_reply[2000];
	
	socket_desc = socket(AF_INET , SOCK_STREAM , 0);
	if (socket_desc == -1)
	{
		printf("Could not create socket");
		return 1;
	}
		
	server.sin_addr.s_addr = inet_addr("192.168.122.140");
	server.sin_family = AF_INET;
	server.sin_port = htons( 12900 );

	if (connect(socket_desc , (struct sockaddr *)&server , sizeof(server)) < 0)
	{
		printf("Could not connect");
		return 1;
	}
	
	message = "GET /system/lbstatus HTTP/1.1\r\nHost: 192.168.122.66:12900\r\nUser-Agent: tcp-tester\r\nAccept: */*\r\n\r\n";
	if( send(socket_desc , message , strlen(message) , 0) < 0)
	{
		printf("Could not send");
		return 1;
	}
	
	// Enough to receive 200 OK, but not even \r and \n
	if( recv(socket_desc, server_reply , 15 , 0) < 0)
	{
		printf("Could not recv");
		return 1;
	}

	printf(server_reply);
	
        // This should trigger the issue; shutdown() would be the proper approach for this.
	close (socket_desc);
	
	return 0;
}

```

**Comment by mikkolehtisalo** [comment]
I was able to reproduce this on 1.3 using a rough C program. However 2.0-alpha5 seems to handle similar situation quietly.

```
#include <stdio.h>
#include <string.h>
#include <sys/socket.h>
#include <arpa/inet.h>
#include <unistd.h>

int main(int argc , char *argv[])
{
	int socket_desc;
	struct sockaddr_in server;
	char *message , server_reply[2000];
	
	socket_desc = socket(AF_INET , SOCK_STREAM , 0);
	if (socket_desc == -1)
	{
		printf("Could not create socket");
		return 1;
	}
		
	server.sin_addr.s_addr = inet_addr("192.168.122.140");
	server.sin_family = AF_INET;
	server.sin_port = htons( 12900 );

	if (connect(socket_desc , (struct sockaddr *)&server , sizeof(server)) < 0)
	{
		printf("Could not connect");
		return 1;
	}
	
	message = "GET /system/lbstatus HTTP/1.1\r\nHost: 192.168.122.140:12900\r\nUser-Agent: tcp-tester\r\nAccept: */*\r\n\r\n";
	if( send(socket_desc , message , strlen(message) , 0) < 0)
	{
		printf("Could not send");
		return 1;
	}
	
	// Enough to receive 200 OK, but not even \r and \n
	if( recv(socket_desc, server_reply , 15 , 0) < 0)
	{
		printf("Could not recv");
		return 1;
	}

	printf(server_reply);
	
        // shutdown() would be the clean way to do this
	close (socket_desc);
	
	return 0;
}
```

**Comment by joschi** [comment]
I can reproduce the issue neither with Graylog 1.3.4-SNAPSHOT (to be released soon) nor with Graylog 2.0.0-alpha.5, so I guess the issue can be closed.

Graylog 1.3.4-SNAPSHOT
```
$ ./tcp-tester
HTTP/1.1 200 OK
$ curl -i http://127.0.0.1:12900/system/lbstatus
HTTP/1.1 200 OK
Content-Type: text/plain
X-Graylog-Node-ID: cd03ee44-b2a7-4824-be16-bb7456149dbd
X-Runtime-Microseconds: 112
Transfer-Encoding: chunked

ALIVE
```
Graylog 2.0.0-alpha.5
```
$ ./tcp-tester
HTTP/1.1 200 OK
$ curl -i http://127.0.0.1:12900/system/lbstatus
HTTP/1.1 200 OK
X-Graylog-Node-ID: cd03ee44-b2a7-4824-be16-bb7456149dbd
X-Runtime-Microseconds: 17300
Content-Type: text/plain
Date: Wed, 16 Mar 2016 11:30:47 GMT
Content-Length: 5

ALIVE
```
